src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,94 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const defaultMovie = {
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  };
+
+  const [count, setCount] = useState(0);
+  const [form, setForm] = useState(defaultMovie);
+
+  const isButtonDisabled = !form.title || !form.imgUrl
+    || !form.imdbUrl || !form.imdbId;
+
+  const handleTitleChange = (newValue: string) => {
+    setForm({ ...form, title: newValue.trimStart() });
+  };
+
+  const handleDescriptionChange = (newValue: string) => {
+    setForm({ ...form, description: newValue.trimStart() });
+  };
+
+  const handleImgUrlChange = (newValue: string) => {
+    setForm({ ...form, imgUrl: newValue.trimStart() });
+  };
+
+  const handleImdbUrlChange = (newValue: string) => {
+    setForm({ ...form, imdbUrl: newValue.trimStart() });
+  };
+
+  const handleImdbIdChange = (newValue: string) => {
+    setForm({ ...form, imdbId: newValue.trimStart() });
+  };
```

Comment:
You can make one handler, which will be the first parameter to accept the name of the field that needs to be updated, and this will greatly shorten the code. 
Also, the data from the form must be taken up-to-date - that is, from the first argument of the callback:
```suggestion
  const handleFormUpdate = (fieldName: string, newValue: string) => {
    setForm((prevFrom) => {
      return {
        ...prevForm,
        [fieldName]: newValue
      }
    });
  };
```